tests/integration/src/test/java/org/mule/test/routing/ScatterGatherRouterTestCase.java

```diff
@@ -273,6 +277,17 @@ public void setsVariablesAfterRouting() throws Exception
         FlowAssert.verify("setsVariablesAfterRouting");
     }
 
+    @Test
+    public void returnsCorrectDataType() throws Exception
+    {
+        MuleEvent event = getTestEvent(TEST_PAYLOAD);
+        event.getMessage().setProperty(CONTENT_TYPE_PROPERTY, "application/json", PropertyScope.INBOUND);
+        MuleMessage response = runFlow("dataType", event).getMessage();
+        assertThat(response, is(Matchers.instanceOf(DefaultMessageCollection.class)));
+        assertThat(((DefaultMessageCollection) response).size(), is(3));
+        assertThat(((DefaultMessageCollection) response).getMessage(0).getDataType().getMimeType(), is("text/plain"));
```

Comment:
shouldn't you also test that the other 2 responses have the same mime Type you set on the original?